NS_Install2.sh

```diff
@@ -84,31 +84,60 @@ sleep 10
 done
 EOF
 
-cs=`grep 'API_SECRET=' /etc/nsconfig | head -1 | cut -f2 -d'"'`
-
-echo "Current API secret is: $cs"
-echo
-echo "If you would like to change it please enter the new secret now or hit enter to leave the same"
+cs2=`grep 'API_SECRET=' /etc/nsconfig | cut -s -f2 -d'"'` # This is API_SECRET if double quotes are used in nsconfig.
+cs1=`grep 'API_SECRET=' /etc/nsconfig | cut -s -f2 -d\'` # This is API_SECRET if single quotes are used in nsconfig.
+cs="$cs2"
+if [ "$cs2" = "" ]
+then
+  cs="$cs1" # This is the current secret (API_SECRET) from nsconfig.
+fi
 
-for j in {1..1000}
+got_it=0
+while [ $got_it -lt 1 ]
 do
-read -t 0.001 dummy
-done
-read -p "New secret 12 character minimum length (blank to skip change) : " ns
+go_back=0
+clear
+exec 3>&1
+Value=$(dialog --colors --ok-label "Submit" --form "       \Zr Developed by the xDrip team \Zn\n\n\n\
+Your current API_SECRET is $cs\n\n\
+You can press escape to maintain the existing one.  Or, enter a new one with at least 12 characters excluding the following.\n\n\
+$ \" \\\n " 19 50 0 "API_SECRET:" 1 1 "$secr" 1 14 25 0 2>&1 1>&3)
```

Comment:
when using single quotes the following are allowed: $ " \ \n the only one that is not allowed is ' it self.

Reply:
I thought Nightscout itself had difficulty with $..Even in Heroku, I thought $ was unacceptable.

Reply:
Considering we still have systems that have " in their nsconfig, how about this for the exceptions?

&  '  "

Reply:
Should also be OK. The most important part is to also remove the ' from the text.

Reply:
I don't know what you mean by that.